pkg/collector/corechecks/windows_event_log/event_formatting.go

```diff
@@ -0,0 +1,224 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build windows
+
+package evtlog
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	agentEvent "github.com/DataDog/datadog-agent/pkg/metrics/event"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api"
+)
+
+func (c *Check) renderEventValues(winevent *evtapi.EventRecord, ddevent *agentEvent.Event) error {
+	// Render the values
+	vals, err := c.evtapi.EvtRenderEventValues(c.systemRenderContext, winevent.EventRecordHandle)
+	if err != nil {
+		return fmt.Errorf("failed to render values: %w", err)
+	}
+	defer vals.Close()
+
+	// Timestamp
+	ts, err := vals.Time(evtapi.EvtSystemTimeCreated)
+	if err != nil {
+		// if no timestamp default to current time
+		ts = time.Now().UTC().Unix()
+	}
+	ddevent.Ts = ts
+	// FQDN
+	fqdn, err := vals.String(evtapi.EvtSystemComputer)
+	if err != nil || c.session == nil {
+		// use DD hostname
+		//   * if collecting from local computer
+		//   * if fail to fetch hostname of remote computer
+		fqdn, _ = hostname.Get(context.TODO())
```

Comment:
Can an empty hostname be used instead here? Would this return a different hostname than the one used by the sender by default?

Reply:
I think we could do that, though other places in the agent use `hostname.Get` instead of relying on the default. However `hostname.Get` implements a cache so it seems like it is intended to be called. It might be worth looking into which method makes more sense.

It looks like the default hostname is initialized when the demultiplexer is created
https://github.com/DataDog/datadog-agent/blob/ecedf4648f41193988b4727fc6f893a0dfc3991e/comp/aggregator/demultiplexer/demultiplexer.go#L47-L55
and then passed to the aggregator, which then passes it to the check sender.

So as long as `hostname.Get` always returns the same thing then the result should be the same if we use an empty hostname here instead. unless `empty_default_hostname` is set, but I'm not familiar with the that use case and how it might affect this.

Reply:
updated to just use empty string